plugins/binanceExchange_ws.go

```diff
@@ -0,0 +1,283 @@
+package plugins
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/adshao/go-binance/v2"
+	"github.com/stellar/kelp/api"
+	"github.com/stellar/kelp/model"
+)
+
+const (
+	STREAMTICKER = "@ticker"
+	TTLTIME      = time.Second * 3 // ttl time in seconds
+)
+
+var (
+	ErrConversionWsMarketEvent = errConversion{from: "interface", to: "*binance.WsMarketStatEvent"}
+)
+
+type errMissingSymbol struct {
+	symbol string
+}
+
+func (err errMissingSymbol) Error() string {
+	return fmt.Sprintf("Symbol %s is missing from exchange intizialization", err.symbol)
+}
+
+type errConversion struct {
+	from string
+	to   string
+}
+
+func (err errConversion) Error() string {
+	return fmt.Sprintf("Error conversion from %s to %s", err.from, err.to)
+}
+
+type stream struct {
+	doneC   chan struct{}
+	stopC   chan struct{}
+	cleanup func()
+}
+
+//Wait until the stream ends
+func (s stream) Wait() {
+
+	if s.doneC == nil {
+		return
+	}
+
+	<-s.doneC
+}
+
+//Close the stream and cleanup any data
+func (s stream) Close() {
+	if s.stopC == nil {
+		return
+	}
+	s.stopC <- struct{}{}
+	s.stopC = nil
+
+	if s.cleanup != nil {
+		s.cleanup()
+	}
+}
+
+type mapData struct {
+	data      interface{}
+	createdAt time.Time
+}
+
+func isStale(data mapData, ttl time.Duration) bool {
+
+	return time.Now().Sub(data.createdAt).Seconds() > ttl.Seconds()
+}
+
+type mapEvents struct {
+	data map[string]mapData
+	mtx  *sync.Mutex
```

Comment:
if we can change this to a `*sync.RWMutex` that is ideal.
Then on Set we can do `rwlock.Lock()` and on Get() we can do `rwlock.RLock()`. This will be more performance than a mutex.